File: app/src/main/java/me/pwcong/simplesidebar/activity/MainActivity.java
package me.pwcong.simplesidebar.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import me.pwcong.simplesidebar.R;
import me.pwcong.simplesidebar.adapter.SimpleAdapter;
import me.pwcong.simplesidebar.conf.Constants;
import me.pwcong.simplesidebar.entity.SimpleData;
import me.pwcong.simplesidebar.view.SimpleSideBar;

/**
 * Created by pwcong on 2016/8/2.
 */
public class MainActivity extends AppCompatActivity {

    LinearLayout linearLayout;
    SimpleSideBar sideBar;
    RecyclerView recyclerView;

    List<SimpleData> simpleDatas;

    SimpleAdapter simpleAdapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        initVariable();

        linearLayout= (LinearLayout) findViewById(R.id.root);

        recyclerView= (RecyclerView) findViewById(R.id.recycler_view);
        recyclerView.setLayoutManager(new LinearLayoutManager(MainActivity.this));

        simpleAdapter=new SimpleAdapter(MainActivity.this, simpleDatas, new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                TextView textView = (TextView) findViewById(R.id.tv_simple);

                Snackbar.make(linearLayout,textView.getText(),Snackbar.LENGTH_SHORT).show();
            }
        });

        recyclerView.setAdapter(simpleAdapter);

        sideBar= (SimpleSideBar) findViewById(R.id.sideBar);

        sideBar.setOnLetterTouchedChangeListener(new SimpleSideBar.OnLetterTouchedChangeListener() {
            @Override
            public void onTouchedLetterChange(String letterTouched) {

                int pos=simpleAdapter.getLetterPosition(letterTouched);
                if(pos!=-1){
                    recyclerView.scrollToPosition(pos);
                }
            }
        });


    }

    private void initVariable(){

        simpleDatas=new ArrayList<>();

        for(int i=0;i<26;i++){

            simpleDatas.add(new SimpleData(Constants.TYPE_HEADER,(char)(i+65)+""));
            for(int j=0;j<5;j++){
                simpleDatas.add(new SimpleData(Constants.TYPE_DATA,(char)(i+65)+" Hello World "+ j));
            }

        }

    }


}

File: app/src/main/java/com/github/pwcong/simplesidebardemo/components/activity/MainActivity.kt
package com.github.pwcong.simplesidebardemo.components.activity

import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.google.android.material.snackbar.Snackbar

import com.github.pwcong.simplesidebar.view.SimpleSideBar
import com.github.pwcong.simplesidebar.view.SimpleSideBar.OnLetterTouchedChangeListener
import com.github.pwcong.simplesidebardemo.R
import com.github.pwcong.simplesidebardemo.adapter.MainAdapter
import github.pwcong.simplesidebardemo.constant.Constant
import com.github.pwcong.simplesidebardemo.entity.MainData


class MainActivity : AppCompatActivity() {
    var linearLayout: LinearLayout? = null
    var sideBar: SimpleSideBar? = null
    var recyclerView: RecyclerView? = null

    var mainDataList: MutableList<MainData>? = null
    var mainAdapter: MainAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        initVariable()

        linearLayout = findViewById<View>(R.id.root) as LinearLayout

        recyclerView = findViewById<View>(R.id.recycler_view) as RecyclerView
        recyclerView!!.layoutManager = LinearLayoutManager(this@MainActivity)

        mainAdapter = MainAdapter(this@MainActivity, mainDataList!!) {
            val textView = findViewById<View>(R.id.tv_simple) as TextView
            Snackbar.make(linearLayout!!, textView.text, Snackbar.LENGTH_SHORT).show()
        }

        recyclerView!!.setAdapter(mainAdapter)

        sideBar = findViewById<View>(R.id.sidebar) as SimpleSideBar
        sideBar!!.setOnLetterTouchedChangeListener(object : OnLetterTouchedChangeListener {
            override fun onTouchedLetterChange(letterTouched: String?) {
                val pos: Int = mainAdapter!!.getLetterPosition(letterTouched)
                if (pos != -1) {
                    recyclerView!!.scrollToPosition(pos)
                }
            }
        })
    }

    private fun initVariable() {
        mainDataList = mutableListOf()

        for (i in 0..25) {
            mainDataList!!.add(
                MainData(
                    Constant.TYPE_HEAD,
                    (i + 65).toChar().toString() + ""
                )
            )
            for (j in 0..4) {
                mainDataList!!.add(
                    MainData(
                        Constant.TYPE_ITEM,
                        (i + 65).toChar().toString() + " Hello World " + j
                    )
                )
            }
        }
    }
}
